blobstore/buckets.go

```diff
@@ -75,17 +76,22 @@ func (s3Ctrl *S3Controller) ListBuckets() (*s3.ListBucketsOutput, error) {
 //	}
 
 type BucketInfo struct {
-	ID   int    `json:"id"`
-	Name string `json:"name"`
+	ID      int    `json:"id"`
+	Name    string `json:"name"`
+	CanRead bool   `json:"can_read"`
 }
 
 func (bh *BlobHandler) HandleListBuckets(c echo.Context) error {
 	var allBuckets []BucketInfo
-	currentID := 1 // Initialize ID counter
+
 	bh.Mu.Lock()
-	for i := 0; i < len(bh.S3Controllers); i++ {
+	defer bh.Mu.Unlock()
+
+	fullAccess := false
```

Comment:
No, this flag indicates that Fine-Grained Access Control (FGAC) checks are unnecessary.  Consider this example in the `list.go` file [link](https://github.com/Dewberry/s3api/blob/feature-read-restriction/blobstore/list.go#L158).  First, we check if the user has full access. If so, the subsequent checks in the function become redundant and resource-intensive, so we skip them. Otherwise, the code continues evaluating the next statement in the boolean OR chain.